packages/react-ui-rent/src/LeadForm/Fields/RequiredField.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import { Field } from '@rentpath/react-ui-core'
+
+@themed(/^RequiredField/)
+
+export default class RequiredField extends Component {
+  static propTypes = {
+    theme: PropTypes.object,
+    name: PropTypes.string,
+    type: PropTypes.string,
+    placeholder: PropTypes.string,
+  }
+
+  static defaultProps = {
+    theme: {},
+  }
+
+  constructor(props) {
+    super(props)
+    this.inputChange = this.inputChange.bind(this)
+    this.state = { labelVisible: true }
+  }
+
+  get fieldLabel() {
+    const { placeholder, theme } = this.props
+
+    if (this.state.labelVisible) {
+      return {
+        text: placeholder,
+        className: theme.RequiredField_Label,
+      }
+    }
+
+    return null
+  }
+
+  inputChange(event) {
+    if (event.target.value) {
+      if (this.state.labelVisible) {
+        this.setState({ labelVisible: false })
+      }
+    } else {
+      this.setState({ labelVisible: true })
+    }
+  }
+
+  render() {
+    const {
+      theme,
+      placeholder,
+      ...props
+    } = this.props
+
+    return (
+      <Field
+        className={theme.RequiredField}
+        theme={theme}
```

Comment:
You don't need to pass this down.

Reply:
I do in this case. `props` vs `this.props`. The `placeholder` is filtered out above and theme is used in `className`

Reply:
That's fine you still don't need it.  Field is already themed.